### PR TITLE
L1Penalty bug fixed

### PR DESCRIPTION
**Pull request by jakezhaojb**
`self.provideOutput` bug fixed.

### CODE REVIEW

**Comment by soumith** [comment]
Thanks jake

File: L1Penalty.lua [hunk 1]
```diff
@@ -4,10 +4,11 @@ local L1Penalty, parent = torch.class('nn.L1Penalty','nn.Module')
 --[gradOutput] to the gradient of the L1 loss. The [input] is copied to 
 --the [output]. 
 
-function L1Penalty:__init(l1weight, sizeAverage)
+function L1Penalty:__init(l1weight, sizeAverage, provideOutput)
     parent.__init(self)
     self.l1weight = l1weight 
     self.sizeAverage = sizeAverage or false  
+    self.provideOutput = provideOutput or true
```
Comment (fmassa) [inline L1Penalty.lua]: Line 11 always evaluate to true, even if provideOutput is false

Comment (soumith) [inline L1Penalty.lua]: you are right. i should not merge PRs from mobile, even if they are super-small!

Comment (soumith) [inline L1Penalty.lua]: fixed in trunk via https://github.com/torch/nn/commit/b74da0080e3f6af53836e36478d6df58c572d351

**Pull request merged**